client/src/components/History/Modals/SelectorModal.vue

```diff
@@ -1,124 +1,248 @@
+const selectedHistories: Ref<HistorySummary[]> = ref([]);
+
+function historyClicked(history: HistorySummary) {
+    if (props.multiple) {
+        const index = selectedHistories.value.indexOf(history);
+        if (index !== -1) {
+            selectedHistories.value.splice(index, 1);
+        } else {
+            selectedHistories.value.push(history);
+        }
+    } else {
+        emit("selectHistory", history);
+        modal.value?.hide();
+    }
+}
+
+function selectHistories() {
+    emit("selectHistories", selectedHistories.value);
+    selectedHistories.value.length = 0;
+    modal.value?.hide();
+}
+
+const router = useRouter();
+const historyStore = useHistoryStore();
+
+function setCurrentHistory(history: HistorySummary) {
+    historyStore.setCurrentHistory(history.id);
+    modal.value?.hide();
+}
+
+function setCenterPanelHistory(history: HistorySummary) {
+    router.push(`/histories/view?id=${history.id}`);
+    modal.value?.hide();
+}
+
+function openInMulti(history: HistorySummary) {
+    router.push("/histories/view_multiple");
+    historyStore.pinHistory(history.id);
+    modal.value?.hide();
+}
+</script>
+
 <template>
-    <b-modal ref="modal" v-bind="$attrs" :title="title | l" footer-class="justify-content-between" v-on="$listeners">
-        <b-form-group :description="'Filter histories' | l">
-            <b-form-input v-model="filter" type="search" :placeholder="'Search Filter' | l" />
-        </b-form-group>
-
-        <b-table
-            striped
-            hover
-            sticky-header="50vh"
-            primary-key="id"
-            :fields="fields"
-            :filter="filter"
-            :items="histories"
-            :per-page="perPage"
-            :current-page="currentPage"
-            :selectable="true"
-            :sort-by.sync="sortBy"
-            :sort-desc.sync="sortDesc"
-            :sort-compare="currentFirstSortCompare"
-            :select-mode="multiple ? 'multi' : 'single'"
-            selected-variant="success"
-            @row-selected="rowSelected"
-            @filtered="onFiltered">
-            <template v-slot:cell(name)="row">
-                {{ row.item.name }} <i v-if="row.item.id === currentHistoryId"><b>(Current)</b></i>
-            </template>
-            <template v-slot:cell(tags)="row">
-                <stateless-tags :value="row.item.tags" :disabled="true" />
+    <div>
+        <b-modal ref="modal" v-bind="$attrs" content-class="history-selector-modal" v-on="$listeners">
+            <template v-slot:modal-title>
+                <Heading h2 inline size="sm">{{ localize(title) }}</Heading>
             </template>
-            <template v-slot:cell(update_time)="data">
-                <UtcDate :date="data.value" mode="elapsed" />
+
+            <b-form-group :description="localize('Filter histories')">
+                <b-form-input v-model="filter" type="search" debounce="400" :placeholder="localize('Search Filter')" />
```

Comment:
I've given it manual testing and I love the new UX!!
However, it seems to be an issue with pagination and filtering. To reproduce:
- Go to page 2
- Try to search for a history you know must be there...
- The expected history is not shown :(

Reply:
Good catch!